src/builder/commands/pip.rs

```diff
@@ -313,24 +314,41 @@ impl BuildStep for Py3Install {
     }
 }
 
+fn parse_req_filename(line: &str) -> Option<&str> {
+    let res =
+        vec!["-r", "--requirement ", "--requirement=",
+             "-c", "--constraint ", "--constraint="]
+            .into_iter()
+            .map(|x| (x, x.len()))
+            .collect::<HashMap<_, _>>();
```

Comment:
You are creating an intermediate vector and a hashmap here. You don't need too. You may iterate over constant array like this: https://is.gd/oyNNsp (and yes, we usually avoid excessive memory allocations in rust)